tests/integration/multithreading_test.py

```diff
@@ -50,3 +56,92 @@ def test_decorator_works_in_a_new_thread():
     assert len(spans) == 2
     assert spans[0]['name'] == 'service1_do_stuff'
     assert spans[1]['name'] == 'index'
+
+
+def _do_one_little_request(output_q, my_input):
+    print('_do_one_little_request(%s): %r' % (my_input, get_default_tracer()))
+    with zipkin_span(
+        service_name='_one_little_request',
+        span_name='do-the-thing',
+        binary_annotations={'input.was': my_input},
+    ) as span_ctx:
+        print('_do_one_little_request(%s): in new span; %r' % (
+            my_input, get_default_tracer()))
+        time.sleep(random.random())
+        span_ctx.add_annotation('time-' + my_input)
+        output_q.put(my_input + '-output')
+
+
+def _do_test_concurrent_subrequests_in_threads(thread_class):
+    """What if a thread has a span context, then fires off N threads to do N
+    subrequests in parallel?
+
+    Surely the spans of the subrequests should be siblings whose parentID is
+    equal to the first thread's span context's spanID.
+
+    Furthermore, a final sub-span created in the first thread should be a child
+    of that thread (no leaking of span stack from the child threads).
+    """
+    transport = MockTransportHandler()
+    with zipkin_span(
+        service_name='main_thread',
+        span_name='handle_one_big_request',
+        transport_handler=transport,
+        sample_rate=100.0,
+        encoding=Encoding.V2_JSON,
+        use_128bit_trace_id=True,
+    ) as span_ctx:
+        print('in root span; main thread: %r' % get_default_tracer())
+        assert True is span_ctx._is_local_root_span
+        assert span_ctx.logging_context
+        expected_trace_id = span_ctx.zipkin_attrs.trace_id
+        expected_parent_id = span_ctx.zipkin_attrs.span_id
+
+        # Now do three subrequests
+        req_count = 3
+        threads = []
+        output_q = queue.Queue()
+        for thread_idx in range(req_count):
+            this_thread = thread_class(
+                target=_do_one_little_request, args=(output_q,
```

Comment:
Can you rebase on the `use_black_for_code_style` branch and run black on these files? `tox -e black` should be enough to automatically update them

Reply:
@drolando This is great!  We switched our huge codebase over to pre-commit & black not too long ago, and we've been loving it.

I haven't mentioned it before, but I've been really impressed and pleased with how easy it's been to keep submissions high quality thanks to the infrastructure you've already got going here:  pre-commit, blocking on 100% line coverage in unit tests, etc.

Enforcing black formatting will make contributing even easier!